src/engraving/libmscore/paste.cpp

```diff
@@ -1059,48 +1071,37 @@ void Score::cmdPaste(const QMimeData* ms, MuseScoreView* view, Fraction scale)
         } else {
             els.append(_selection.elements());
         }
-
+        Element* newEl = 0;
         for (Element* target : els) {
+            if (!target->selected()) { // might be deselected/removed by a previous operation
+                continue;
+            }
             el->setTrack(target->track());
-            Element* nel = el->clone();
+            newEl = el->clone();
             addRefresh(target->abbox());         // layout() ?!
             EditData ddata(view);
-            ddata.dropElement = nel;
+            ddata.dropElement = newEl;
             if (target->acceptDrop(ddata)) {
                 if (el->isNote()) {
-                    // dropping a note replaces and invalidates the target,
-                    // so we need to deselect it
-                    ElementType targetType = target->type();
-                    deselect(target);
-
-                    // perform the drop
-                    target->drop(ddata);
-
-                    TDuration durationType(duration);
-                    if (target->isChordRest()) {
-                        ChordRest* targetCR = dynamic_cast<ChordRest*>(target);
-                        score()->changeCRlen(targetCR, durationType);
-                    } else if (target->isNote()) {
-                        ChordRest* targetCR = dynamic_cast<Note*>(target)->chord();
-                        score()->changeCRlen(targetCR, durationType);
-                    }
-
-                    // if the target is a rest rather than a note,
-                    // a new note is generated, and nel becomes invalid as well
-                    // (ChordRest::drop() will select it for us)
-                    if (targetType == ElementType::NOTE) {
-                        select(nel);
+                    // duration information needs to be provided separately in this case
+                    if (target->isNote() && toNote(target)->chord()->ticks() != duration) {
+                        newEl = replaceNote(toNote(target)->chord(), toNote(newEl), duration);
+                    } else if (target->isChordRest()) {
+                        newEl = replaceNote(toChordRest(target), toNote(newEl), duration);
+                    } else {
+                        target->drop(ddata);
```

Comment:
would prefer that above logic was all inside Note::drop( ) and ChordRest::drop( ) but would require adding even more info (duration) to the already-bloated EditData structure...